pkgs/by-name/gp/gpt4all/package.nix

```diff
@@ -86,6 +92,10 @@ stdenv.mkDerivation (finalAttrs: {
     # https://github.com/nomic-ai/gpt4all/issues/3468
     (lib.cmakeBool "LLMODEL_KOMPUTE" false)
   ]
+  ++ lib.optionals cudaSupport [
+    "-DCUDA_HOST_COMPILER=${gcc12Stdenv.cc}/bin/c++"
```

Comment:
have you tried using `cudaPackages.backendStdenv.cc` ?

Reply:
Nevermind, it explodes in the same way. Reverting!